packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -189,38 +197,42 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
     };
 
     let onFocus = (e) => {
-      // If the focused element is in the current scope, and not in the active scope,
-      // update the active scope to point to this scope.
-      let isInScope = isElementInScope(e.target, scopeRef.current);
-      if (isInScope && (!activeScope || !isElementInScope(e.target, activeScope.current))) {
-        activeScope = scopeRef;
-      }
+      e.stopPropagation();
+      activeScope = scopeRef;
+      focusedNode.current = e.target;
+    };
 
-      // Save the currently focused node in this scope
-      if (isInScope) {
-        focusedNode.current = e.target;
-      }
+    let onBlur = (e) => {
+      e.stopPropagation();
+      let isInAnyScope = isElementInAnyScope(e.relatedTarget, scopesMap);
 
-      // If a focus event occurs outside the active scope (e.g. user tabs from browser location bar),
-      // restore focus to the previously focused node or the first tabbable element if none.
-      if (activeScope === scopeRef && !isInScope) {
-        if (focusedNode.current) {
-          focusedNode.current.focus();
-        } else {
-          focusFirstInScope(scopeRef.current);
-        }
+      if (!isInAnyScope) {
+        activeScope = scopeRef;
+        focusedNode.current = e.target;
```

Comment:
Looks like you could get rid of `focusedNode` entirely. It's not used anywhere else.

Reply:
using it now for the document level focusin events, thanks for the catch